drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/widget/analysis/checks/RangeCheck.java

```diff
@@ -0,0 +1,193 @@
+    }
+
+    private String invert( final String operator ) {
+
+        switch ( operator ) {
+            case "==":
+                return "!=";
+            case "!=":
+                return "==";
+            case ">":
+                return "<=";
+            case "<":
+                return ">=";
+            case ">=":
+                return "<";
+            case "<=":
+                return ">";
+            default:
+                return operator;
+        }
+    }
+
+    private RuleInspectorClone makeClone() {
+        return new RuleInspectorClone( ruleInspector.getRule(),
+                                       ruleInspector.getCache() );
+    }
+
+    @Override
+    public Issue getIssue() {
+        final Issue issue = new Issue( Severity.NOTE,
+                                       AnalysisConstants.INSTANCE.MissingRangeTitle(),
+                                       ruleInspector.getRowIndex() + 1 );
+
+        issue.getExplanation()
+             .addParagraph( AnalysisConstants.INSTANCE.MissingRangeP1( ruleInspector.getRowIndex() + 1 ) );
+
+        return issue;
+    }
+
+    private class RuleInspectorClone
+            extends RuleInspector {
+
+        private final InspectorList<ConditionsInspector> conditionsInspectors;
+        boolean containsInvertedItems = false;
+
+        public RuleInspectorClone( final Rule rule,
+                                   final RuleInspectorCache cache ) {
+            super( rule,
+                   cache );
+            conditionsInspectors = makeConditionsInspectors();
+        }
+
+        @Override
+        public InspectorList<ConditionsInspector> getConditionsInspectors() {
+            return conditionsInspectors;
+        }
+
+        private InspectorList<ConditionsInspector> makeConditionsInspectors() {
+            final InspectorList<ConditionsInspector> clonedConditionInspectors = new InspectorList<>();
+
+            for ( final ConditionsInspector original : super.getConditionsInspectors() ) {
+
+                final ConditionsInspector clone = new ConditionsInspector();
+
+                for ( final Field field : original.keys() ) {
+
+                    for ( final ConditionInspector originalInspector : original.get( field ) ) {
+                        clone.put( field,
+                                   resolveInspector( originalInspector ) );
+                    }
+                }
+
+                clonedConditionInspectors.add( clone );
+            }
+
+            return clonedConditionInspectors;
+        }
+
+        private ConditionInspector resolveInspector( final ConditionInspector originalInspector ) {
+
+            if ( originalInspector.getCondition() instanceof FieldCondition ) {
+                final FieldCondition fieldCondition = ( FieldCondition ) originalInspector.getCondition();
+
+                if ( fieldCondition.getField().getFieldType().equals( "Integer" ) ) {
```

Comment:
Yep. Long is missing. Made a ticket. While this is missing for Long it does not cause false reports. Our V&V coverage just is not as good as it could be.
https://issues.jboss.org/browse/GUVNOR-2576

Another with special treatment is boolean. Since x == true and x != false are equal.

For values with decimals or any "ranged" comparable we use check the operator and then we make use of the Comparator interface.